tests/goal/test_goal.py

```diff
@@ -80,6 +80,30 @@ def test_goal_simple_args() -> None:
     verify(result.output)
 
 
+@pytest.mark.usefixtures("proc_mock")
+def test_goal_simple_args_with_input_file() -> None:
+    result = invoke("goal clerk compile approval.teal")
```

Comment:
with proc_mock you are probably mocking the execution of the docker itself but remaining bits arent mocked (which makes sense). So id suggest to have a real simple text file (doesn't even have to be teal) but it needs to exist on the cwd path if you want to test:
- paths params being changed by pre processing and post processing
- input file being copied to and from `mocked` volume mount folder. You can mock the `out` file to be returned back to return some other dummy text file (since you arent gonna run actual goal clerk)